lib/hedwig_mopidy.ex

```diff
@@ -74,4 +74,113 @@ defmodule HedwigMopidy do
   def user(message) do
     if is_map(message.user) do message.user.name else message.user end
   end
-end
+
+  def default_playlist do
+    Application.get_env(:hedwig_mopidy, :default_playlist) || "spotify:user:labzeroinnovations:playlist:64mMWs2NiiguFZWtODm6Jh"
+  end
+
+  def favorites_playlist do
+    Application.get_env(:hedwig_mopidy, :favorites_playlist) || "spotify:user:labzeroinnovations:playlist:00l7ibuNDlOGRvfReyaUge"
+  end
+
+  def rejects_playlist do
+    Application.get_env(:hedwig_mopidy, :rejects_playlist) || "spotify:user:labzeroinnovations:playlist:6TO5iMRRgTGvxEFUWI4D89"
+  end
+
+  defmodule Spotify do
+    def get_token do
+      with {:ok, response} <- HTTPoison.post("https://accounts.spotify.com/api/token",
+                                             "grant_type=refresh_token&refresh_token=#{get_refresh_token}",
+                                             ["Authorization": "Basic #{:base64.encode(get_secrets)}",
+                                              "Content-Type": "application/x-www-form-urlencoded"],
+                                             hackney: [pool: :tracklist]) do
+        Poison.decode!(response.body)["access_token"]
+      end
+    end
+
+    def favorite_track(track_uri) do
+      add_track_to_playlist(HedwigMopidy.favorites_playlist, track_uri)
+    end
+
+    def reject_track(playlist_uri, track_uri) do
+      remove_track_from_playlist(playlist_uri, track_uri)
+      add_track_to_playlist(HedwigMopidy.rejects_playlist, track_uri)
+    end
+
+    def add_track_to_playlist(playlist_uri, track_uri) do
+      api_url = transform_playlist_uri(playlist_uri)
+      with {:ok, response} <- HTTPoison.post("#{api_url}?uris=#{track_uri}",
+                                             "",
+                                             ["Authorization": "Bearer #{get_token}",
+                                              "Accept": "application/json"],
+                                             hackney: [pool: :tracklist]) do
+        Poison.decode!(response.body)["snapshot_id"]
+      end
+    end
+
+    def remove_track_from_playlist(playlist_uri, track_uri) do
+      api_url = transform_playlist_uri(playlist_uri)
+      with {:ok, response} <- HTTPoison.request!(:delete,
```

Comment:
`HTTPoison.request!/5` will raise an exception, you probably want just `request/5`